src/EnergyPlus/UnitarySystem.cc

```diff
@@ -0,0 +1,4202 @@
+                    }
+
+                    std::string loc_coolingCoilName("");
+                    if (fields.find("cooling_coil_name") != fields.end()) { // not required field
+                        loc_coolingCoilName = UtilityRoutines::MakeUPPERCase(fields.at("cooling_coil_name"));
+                    }
+
+                    std::string loc_ISHundredPercentDOASDXCoil("No");
+                    if (fields.find("use_doas_dx_cooling_coil") != fields.end()) { // not required field, has default
+                        loc_ISHundredPercentDOASDXCoil = UtilityRoutines::MakeUPPERCase(fields.at("use_doas_dx_cooling_coil"));
+                    }
+
+                    Real64 loc_designMinOutletTemp(2.0);
+                    if (fields.find("minimum_supply_air_temperature") != fields.end()) { // not required field, has default
+                        loc_designMinOutletTemp = fields.at("minimum_supply_air_temperature");
+                    }
+
+                    std::string loc_latentControlFlag("SensibleOnlyLoadControl");
+                    if (fields.find("latent_load_control") != fields.end()) { // not required field, has default
+                        loc_latentControlFlag = UtilityRoutines::MakeUPPERCase(fields.at("latent_load_control"));
+                    }
+
+                    std::string loc_suppHeatCoilType("");
+                    if (fields.find("supplemental_heating_coil_object_type") != fields.end()) { // not required field
+                        loc_suppHeatCoilType = UtilityRoutines::MakeUPPERCase(fields.at("supplemental_heating_coil_object_type"));
+                    }
+
+                    std::string loc_suppHeatCoilName("");
+                    if (fields.find("supplemental_heating_coil_name") != fields.end()) { // not required field
+                        loc_suppHeatCoilName = UtilityRoutines::MakeUPPERCase(fields.at("supplemental_heating_coil_name"));
+                    }
+
+                    std::string loc_coolingSAFMethod("");
+                    if (fields.find("cooling_supply_air_flow_rate_method") != fields.end()) { // not required field
+                        loc_coolingSAFMethod = UtilityRoutines::MakeUPPERCase(fields.at("cooling_supply_air_flow_rate_method"));
+                    }
+
+                    Real64 loc_coolingSAFMethod_SAFlow(-999.0);
+                    if (fields.find("cooling_supply_air_flow_rate") != fields.end()) { // not required field, autosizable
+                        std::string tempFieldVal = UtilityRoutines::MakeUPPERCase(fields.at("cooling_supply_air_flow_rate"));
+                        if (UtilityRoutines::SameString(tempFieldVal, "AutoSize")) {
+                            loc_coolingSAFMethod_SAFlow = DataSizing::AutoSize;
+                        } else {
+                            loc_coolingSAFMethod_SAFlow = fields.at("cooling_supply_air_flow_rate");
+                        }
+                    }
```

Comment:
@Myoldmopar this seems like a lot of overhead in getInput functions. Wouldn't it be better to write one or more functions in InputProcessor that handle all the subtleties of getting JSON input data?

One function:
getJSONInput(enum fieldType (e.g., alpha or numeric), std::string fieldString);

Multiple functions:
getJSONAlphaInput(std::string fieldString);
getJSONNumericInput(std:string fieldString);

Also need a function to return the name of the field:
getJSONFieldName(std::string fieldString);

AirLoopHVAC:UnitarySystem,
  A2,  \field Control Type

    ShowWarningMessage("When "+ getJSONFieldName("control_type") + " = " + getJSONAlphaInput("control_type") + " blah blah ";

Not sure how this will work with the new warning message effort.